text/0016-standard-modes.md

```diff
@@ -0,0 +1,357 @@
+- Modes logically shouldn't be in the `HEARTBEAT` - extending this with standard modes is compounding the issue.
+- Having a separate message means that this is completely separate/orthogonal to the existing implementation.
+
+We might also infer the current standard mode from the existing base and custom mode fields in the `HEARTBEAT`.
+This would mean that the message would not have to change, but would have the following implications:
+- Requires caching the mapping to custom modes, which scales by number of autopilots in the system.
+- Requires a discrete custom mode for every standard mode.
+  For ArduPilot and PX4 this means a custom mode would have to be defined for the different return options of rally points vs RTL.
+
+
+## Setting modes: Set standard mode using mapped base/custom modes
+
+We might instead set desired standard mode using the mapped custom/based modes in [MAV_CMD_DO_SET_MODE](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_MODE).
+
+This is not considered desirable because:
+- This would imply there _must_ always be a specific custom mode for each standard mode.
+- It is more complicated for GCS/SDKs, which will have to maintain a record of the mapping between modes.
+  Maybe not a problem for just one node, but scales badly.
+- Overall it is simpler for all users for the mapping to be provided by the stack.
+
+## Setting modes: Updated generic mode setter
+
+The proposed setter command (`MAV_CMD_DO_SET_STANDARD_MODE`) just sets the standard mode, while the old command (`MAV_CMD_DO_SET_MODE`) sets base and custom modes.
+
+We can't just use [MAV_CMD_DO_SET_MODE.param4](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_MODE) to set the standard mode as a system that does not understand this service will ignore the value and attempt to set the base/custom mode.
+
+But we could make this new setter command more generic `MAV_CMD_DO_SET_MODE_V2` and include custom and base modes in it too:
+```xml
+      <entry value="262" name="MAV_CMD_DO_SET_MODE_V2" hasLocation="false" isDestination="false">
+        <description>Set a standard or custom mode.
+          If the mode is not supported the vehicle should ACK with MAV_RESULT_FAILED.
+        </description>
+        <param index="1" label="Standard Mode" enum="MAV_STANDARD_MODE">Standard mode to set. Set to MAV_CMD_DO_SET_STANDARD_MODE if setting a custom mode.</param>
+        <param index="2" label="base mode">Base mode. Ignored if standard mode (param1) is non-zero.</param>
+        <param index="3" label="custom mode">Custom mode. Ignore if standard mode (param1) is non-zero.</param>
+        <param index="4" reserved="true" default="0"/>
+        <param index="5" reserved="true" default="0"/>
+        <param index="6" reserved="true" default="0"/>
+        <param index="7" reserved="true" default="NaN"/>
+      </entry>
+```
+
+## Getting Available modes: Component Metadata rather than Message
```

Comment:
@auturgy @LorenzMeier This section was added suggesting using component information for getting available modes rather than having messages.

Component information is a much better approach - this sort of thing is why the service was invented. But we can't build on that unless it is part of the standard.